service/matching/matchingEngine.go

```diff
@@ -105,6 +108,7 @@ func NewEngine(taskManager persistence.TaskManager,
 	logger log.Logger,
 	metricsClient metrics.Client,
 	domainCache cache.DomainCache,
+	monitor membership.Monitor,
```

Comment:
can you take ServiceResolver as input instead ? Looks like top level resource object has a GetMatchingServiceResolver() method already